src/odfesqlodbc/es_statement.cpp

```diff
@@ -148,6 +151,36 @@ RETCODE ExecuteStatement(StatementClass *stmt, BOOL commit) {
     return CleanUp();
 }
 
+SQLRETURN GetNextResultSet(StatementClass *stmt) {
+    if (stmt == NULL) {
+        return SQL_ERROR;
+    }
+    ConnectionClass *conn = SC_get_conn(stmt);
+    QResultClass *q_res = SC_get_Result(stmt);
+    if ((q_res == NULL) && (conn == NULL)) {
+        return SQL_ERROR;
+    }
+    SQLSMALLINT total_columns = -1;
+    SQLNumResultCols(stmt, &total_columns);
+    if (total_columns == -1) {
+        return SQL_ERROR;
+    }
+    ESResult *es_res = ESGetResult(conn->esconn);
+    while (es_res != NULL) {
```

Comment:
I'm confused as to what this does. It _appears_ to run through all available results and set them to some saved state. But the wording here looks like it's _setting_ instead of _adding_ so if there were multiple results wouldn't we end up only saving the last one?

Reply:
When we are getting data from the server, we are adding it to the queue. This loop is retrieving all results from that queue and appending those results in q_res using CC_Assign_Table_Data. Since we are not overriding and appending at the end, we will have all results, not the last one only.

Reply:
Also, we are saving only last cursor_id to get further result pages from the server. So we are overriding server cursor info since we only need last.

Reply:
Can the names be updated a bit, and comments be added, to reflect the appending of data vs. the assignment (which typically means overwriting)?

Reply:
Yes, I will update

Reply:
done